operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java

```diff
@@ -141,12 +141,31 @@ Integer getAsPort() {
         .getLocalAdminProtocolChannelPort();
   }
 
+  /**
+   * Check if the server is listening on a secure port. NOTE: If the targetted server is a managed
+   * server, this method is overridden to check if the managed server has a secure listen port rather
+   * than the admin server. See PodHelper.ManagedPodStepContext
+   *
+   * @return true if server is listening on a secure port
+   */
   boolean isLocalAdminProtocolChannelSecure() {
     return domainTopology
         .getServerConfig(domainTopology.getAdminServerName())
         .isLocalAdminProtocolChannelSecure();
   }
 
+  /**
+   * Check if the admin server is listening on a secure port.
+   *
+   * @return true if admin server is listening on a secure port
+   */
+  private boolean isAdminServerProtocolChannelSecure() {
+    return domainTopology
+        .getServerConfig(domainTopology.getAdminServerName())
+        .isLocalAdminProtocolChannelSecure();
+  }
+
+
```

Comment:
This looks exactly the same as isLocalAdminProtocolChannelSecure().  Does that mean there's a bug in that method?

Reply:
My guess is that isLocalAdminProtocolChannelSecure should be calling 'getServerName()' instead of 'domainTopology.getAdminServerName()'.  

I know that's not directly related to this pull, but since the patient is open, maybe you can investigate?

Reply:
I don't think isLocalAdminProtocolChannelSecure() is buggy.  If you look at its method comment, it is overriden by PodHelper.ManagedPodStepContext([.isLocalAdminProtocolChannelSecure()]).  So the reported bug manifests when the domain wide admin port is not enabled and the admin server has an SSL port enabled which then results in the "http://admin_server_name:admin_server_ssl_port" AdminURL.

I agree having isLocalAdminProtocolChannelSecure call getServerName() seems to flow more naturally.  The managed server pod will call its overriding version anyway, and the admin server pod will get its own config.  This is an existing method and it works so I did not focus on that aspect.  I will double check this.

Reply:
And one advantage of having isLocalAdminProtocolChannelSecure call getServerName() is that the method then does not need to be overridden in PodHelper.ManagedPodStepContext.